### DISCUSSION TITLE
rh-icon discussion

### DISCUSSION THREAD

**Discussion by heyMP** [RedHat-UX/red-hat-design-system#399] [General]
This is the home for all things rh-icon discussion.  "Do we need our own icon set?", "Are we limiting icon usage?", and more!

pfe-icon page: https://patternflyelements.com/components/icon/

**Comment by bennypowers** [comment]
PFv4 uses font-awesome
PFE uses `fetch` to load arbitrary icon sets, but comes with a default
I think RHDS should use named lit `svg` tagged template literals, which would be `import`ed, eagerly or lazily, depending on the use.

PROS:
- smaller than font-awesome
- easy for developers to handle

CONS:
- JS-only
- less flexible than PFE's approach (but maybe this is a PRO)

**Comment by bennypowers** [comment]
This is coming up in rh-cta, which has a video sub-variant that necessitates and `icon` attr, and presumably an rh-icon

One thing i'd like for `<rh-icon>` is to remove the requirement of prefixing every icon name.

so for example:
```html
<pfe-icon icon="rh-leaf">
```
I want to get rid of the `rh-` part.

*BUT WAIT* you say, What about custom icon sets?


Well, how about:

```html
<rh-icon icon="user" set="far">
```

set defaults to rh, and like pfe-icon v1, can take any string registered as an icon set name.

Thus, pfe-icon 2 (the upcoming 1:1 variant) would register the pf icon set as default, and allow for easy registering of fas and far as well, whereas rh-icon would register the rh icon set by default, and be able to import fas and far from pfe.

**Reply by bennypowers** [reply]
@mwcz wdyt? the icon loading was originally your jam. So how does
a. importing SVGResult objects
b. bouncing iocn set to a separate attr
strike you?
